examples/examples_nodejs_test.go

```diff
@@ -119,3 +135,93 @@ func getJSBaseOptions(t *testing.T) integration.ProgramTestOptions {
 
 	return baseJS
 }
+
+// retryFunc retries a function every 3 seconds for up to 1 minute
+func retryFunc(t *testing.T, fn func() bool) error {
+	t.Helper()
+	ticker := time.NewTicker(time.Second * 3)
+	timeout := time.After(time.Minute * 1)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("timeout after 1 minute")
+		case tc := <-ticker.C:
+			t.Logf("tick: %s", tc.String())
+			if fn() {
+				return nil
+			}
+		}
+	}
+}
+
+// isConnected validates that the $connect lambda was triggered successfully by
+// ensuring there is a record in the dynamodb table
+// It shouldn't be necessary to retry this, but my testing was very inconsistent, sometimes the `Dial`
+// triggered the connect and sometimes it didn't (according to the docs it shouldn't be possible to establish
+// a connection without triggering $connect)
+func isConnected(ctx context.Context, t *testing.T, table string, c *websocket.Conn, client *dynamodb.Client, canRetry bool) error {
```

Comment:
I thought this was going to be easy, but it ended up being extremely annoying/difficult 😖

It might not even be worth it at this point...